Primal/Common/Nostr/NostrObject+Extra.swift

```diff
@@ -252,18 +252,27 @@ fileprivate func createNostrRepostEvent(_ nostrContent: NostrContent) -> NostrOb
 }
 
 fileprivate func createNostrReplyEvent(_ content: String, post: PrimalFeedPost, mentionedPubkeys: [String]) -> NostrObject? {
-    let e = ["e", post.id, RelayHintManager.instance.getRelayHint(post.id), "reply"]
-    let p = ["p", post.pubkey]
-    
-    var root = ["e", post.id, RelayHintManager.instance.getRelayHint(post.id), "root"]
-    for tag in post.tags {
-        if tag.last == "root" {
-            root = tag
-        }
+    var allTags: [[String]] = []
+
+    /// The `e` tags are ordered at best effort to support the deprecated method of positional tags to maximize backwards compatibility
+    /// with clients that support replies but have not been updated to understand tag markers.
+    ///
+    /// https://github.com/nostr-protocol/nips/blob/master/10.md
+    ///
+    /// The tag to the root of the reply chain goes first.
+    /// The tag to the reply event being responded to goes last.
+    if let root = post.tags.last(where: { tag in tag[3] == "root" }) {
```

Comment:
This can lead to a crash as we can't know for sure that tag array has at least 3 members, need to use either ".last" like it was before or "[safe: 3]" subscript to make this code safe.

Reply:
Thanks for the feedback. `.last` is incorrect because it's not guaranteed that the marker is the last element. We can't blindly trust that any responses from any relay will conform to the spec. I've replaced it with `[safe: 3]`.